File: TestAndroidCommonFrame/src/main/java/com/test/androidcommonframe/base/BaseActivity.java
package com.test.androidcommonframe.base;

import android.lorenwang.commonbaseframe.AcbflwBaseActivity;
import android.lorenwang.tools.app.AtlwToastHintUtils;

import com.test.androidcommonframe.R;
import com.test.androidcommonframe.dialog.LoadingDialog;

import org.jetbrains.annotations.Nullable;

import javabase.lorenwang.dataparse.JdplwJsonUtils;
import javabase.lorenwang.tools.common.JtlwCheckVariateUtils;
import kotlinbase.lorenwang.tools.common.bean.KttlwBaseNetResponseBean;

/**
 * 功能作用：基础Activity
 * 创建时间：2020-07-23 11:20 上午
 * 创建人：王亮（Loren wang）
 * 思路：
 * 方法：
 * 注意：
 * 修改人：
 * 修改时间：
 * 备注：
 *
 * @author 王亮（Loren wang）
 */
public abstract class BaseActivity extends AcbflwBaseActivity {
    private LoadingDialog loadingDialog;

    @Override
    public void hideBaseLoading() {
        if (loadingDialog != null) {
            loadingDialog.dismiss();
        }
    }

    @Override
    protected void addContentView(int resId) {
        super.addContentView(resId, R.layout.title_bar_head_view_type_1);
    }

    /**
     * 显示空数据
     */
    protected void showEmptyData() {
        super.showEmptyData(R.layout.empty_data_default);
    }

    @Override
    public void showBaseLoading(boolean allowLoadingBackFinishPage) {
        if (loadingDialog == null) {
            loadingDialog = new LoadingDialog(this);
        }
        if (!loadingDialog.isShowing()) {
            loadingDialog.show(allowLoadingBackFinishPage);
        }
    }

    @Override
    public void userLoginStatusError(@Nullable Object code, @Nullable String message) {
        hideBaseLoading();
    }

    @Override
    public void netReqFail(int netOptionReqCode, @Nullable Object code, @Nullable String message) {
        if (JtlwCheckVariateUtils.getInstance().isNotEmpty(message)) {
            KttlwBaseNetResponseBean bean = JdplwJsonUtils.fromJson(message,
                    KttlwBaseNetResponseBean.class);
            if (bean != null) {
                if (JtlwCheckVariateUtils.getInstance().isNotEmpty(bean.getStateMessage())) {
                    AtlwToastHintUtils.getInstance().toastMsg(bean.getStateMessage());
                }
            } else {
                AtlwToastHintUtils.getInstance().toastMsg(message);
            }
        }
    }

    /**
     * 网络请求成功
     *
     * @param data             响应数据
     * @param netOptionReqCode 网络操作请求code
     */
    @Override
    public <T> void netReqSuccess(int netOptionReqCode, T data) {
    }
}

File: TestAndroidAll/src/main/java/com/example/testapp/base/BaseActivity.kt
package com.example.testapp.base

import android.graphics.Color
import android.lorenwang.commonbaseframe.AcbflwBaseActivity
import android.lorenwang.customview.dialog.AvlwLoadingDialogType1
import android.lorenwang.tools.app.AtlwToastHintUtil
import android.view.ViewStub
import javabase.lorenwang.dataparse.JdplwJsonUtils
import javabase.lorenwang.tools.common.JtlwCheckVariateUtils
import kotlinbase.lorenwang.tools.common.bean.KttlwBaseNetResponseBean

/**
 * 创建时间：2019-04-15 下午 15:15:55
 * 创建人：王亮（Loren wang）
 * 功能作用：
 * 思路：
 * 方法：
 * 注意：
 * 修改人：
 * 修改时间：
 * 备注：
 */
abstract class BaseActivity : AcbflwBaseActivity() {
    private var loadingDialog: AvlwLoadingDialogType1? = null//加载中弹窗
    private lateinit var vsbContent: ViewStub

    override fun showBaseLoading(allowLoadingBackFinishPage: Boolean) {
        if (loadingDialog == null) {
            loadingDialog = AvlwLoadingDialogType1(this)
            loadingDialog?.setProgressBarColor(Color.BLUE);
        }
        if (!loadingDialog!!.isShowing()) {
            loadingDialog?.show()
        }
    }

    override fun hideBaseLoading() {
        if (loadingDialog != null && loadingDialog!!.isShowing()) {
            loadingDialog?.dismiss()
        }
    }

    /**
     * 用户登陆状态异常
     */
    override fun userLoginStatusError(code: Any?, message: String?) {
        hideBaseLoading()
    }

    /**
     * 网络请求成功
     * @param data 响应数据
     * @param netOptionReqCode 网络操作请求code
     */
    override fun <T> netReqSuccess(netOptionReqCode: Int, data: T) {
    }

    /**
     * 网络请求失败
     * @param netOptionReqCode 网络操作请求code
     * @param message 错误信息
     */
    override fun netReqFail(netOptionReqCode: Int, message: String?) {
        if (JtlwCheckVariateUtils.getInstance().isNotEmpty(message)) {
            val bean = JdplwJsonUtils.fromJson(message,
                    KttlwBaseNetResponseBean::class.java)
            if (bean != null) {
                if (JtlwCheckVariateUtils.getInstance().isNotEmpty(bean.stateMessage)) {
                    AtlwToastHintUtil.getInstance().toastMsg(bean.stateMessage)
                }
            } else {
                AtlwToastHintUtil.getInstance().toastMsg(message)
            }
        }
    }


}
